opm/autodiff/FlowMain.hpp

```diff
@@ -328,13 +325,53 @@ namespace Opm
 
 
 
+
+        void setupLogging()
+        {
+            std::string deck_filename = param_.get<std::string>("deck_filename");
+            // create logFile
+            using boost::filesystem::path; 
+            path fpath(deck_filename);
+            std::string baseName;
+            std::string debugFile;
+            if (boost::to_upper_copy(path(fpath.extension()).string()) == ".DATA") {
+                baseName = path(fpath.stem()).string();
+            } else {
+                baseName = path(fpath.filename()).string();
+            }
+            if (param_.has("output_dir")) {
```

Comment:
Since here already have the ``output_dir``, I suggest keeping the function call order and add a ``const std::string output_dir = param_.getDefault()`` to avoid missing any output parameter which should in DEBUG file.